### PR TITLE
Change Logic of Obtaining Recurring Orders Based on Frequency

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/6826

~~Related Front-end PR: https://github.com/opencollective/opencollective-frontend/pull/8992~~

image

### CODE REVIEW

File: server/graphql/v2/enum/ContributionFrequency.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export const GraphQLContributionFrequency = new GraphQLEnumType({
     MONTHLY: {},
     YEARLY: {},
     ONETIME: {},
+    RECURRING: {},
```
Comment (hdiniz) [inline server/graphql/v2/enum/ContributionFrequency.ts:11]: This enum is used in other queries and mutation inputs. For this I think we should filter with another value in the OrderCollectionQuery.

File: server/graphql/v2/query/collection/OrdersCollectionQuery.ts [hunk 2]
```diff
@@ -226,11 +226,13 @@ export const OrdersCollectionResolver = async (args, req: express.Request) => {
...
-      include.push({ model: models.Subscription, required: true, where: { interval: 'month' } });
+      where['interval'] = 'month';
     } else if (args.frequency === 'YEARLY') {
-      include.push({ model: models.Subscription, required: true, where: { interval: 'year' } });
+      where['interval'] = 'year';
+    } else if (args.frequency === 'RECURRING') {
+      where['interval'] = { [Op.in]: ['year', 'month'] };
     }
   } else if (args.onlySubscriptions) {
     include.push({ model: models.Subscription, required: true });
```
Comment (hdiniz) [inline server/graphql/v2/query/collection/OrdersCollectionQuery.ts:238]: Maybe an alternative would be to make `onlySubscriptions` also return processing subscriptions. 

```suggestion
    include.push({ model: models.Subscription, required: false });
    where[Op.and].push({
       [Op.or]: [
         { ['$subscription.id$']: { [Op.ne]: null }],
         { "interval": { [Op.in]: ['year', 'month'] }, "status": "PROCESSING" }
       ]
    })
```

File: server/graphql/v2/enum/ContributionFrequency.ts (hunk 1 above)
Comment (SudharakaP) [reply server/graphql/v2/enum/ContributionFrequency.ts]: Makes sense. I've removed this and used the alternative of making `onlySubscriptions` also return processing subscriptions. Thanks. 👍🏽

File: server/graphql/v2/query/collection/OrdersCollectionQuery.ts (hunk 2 above)
Comment (SudharakaP) [reply server/graphql/v2/query/collection/OrdersCollectionQuery.ts]: Done. 👍🏽

File: server/graphql/schemaV2.graphql [hunk 3]
```diff
@@ -944,6 +944,7 @@ enum ContributionFrequency {
   MONTHLY
   YEARLY
   ONETIME
+  RECURRING
```
Comment (hdiniz) [inline server/graphql/schemaV2.graphql:947]: I think this value was left over, right?

**Review by hdiniz — APPROVED** [review]

File: server/graphql/schemaV2.graphql (hunk 3 above)
Comment (SudharakaP) [reply server/graphql/schemaV2.graphql]: Ops, that seems to be the case. I've corrected it. Will merge ones test pass. Thanks for the reviews. 👍🏽

**Pull request merged**